### DISCUSSION TITLE
Is there any method to change the chain in opensea and uniswap dapp's before connecting with walletconnect?

### DISCUSSION THREAD

**Discussion by subhathLiveroom** [WalletConnect/walletconnect-monorepo#3241] [Developer Support]
Is there any way give access only to the polygon chain when connecting the wallet to the dapp, at the time of integrating the walletconnect in developing the wallet?
I found that, we can change the chain of walletconnect demo app before connecting. But, couldn't find a method to do so in opensea or uniswap.

we keep user's account using 4337 contract and that 4337 contract address as the user's account address.In that case we use 4337 contract address to connect dapps like opensea through walletconnect.When connecting to opensea at first time, opensea asks for personal sign, although it asks personal sign through etherium chain but we have only deployed the contract in polygon chain. Because of that, we need to find a way to change the chain before connecting to the Opensea or else need to find a way to request or decide the chain from the wallet side. Otherwise we have to deploy the 4337 contract in Ethereum chain.We have to decide the chain before connecting to the Dapp in Walletconnect demo Dapp (Link : https://react-app.walletconnect.com/ ) but that option isn't in the Dapps like Opensea and Uniswap.

**Comment by glitch-txs** [comment]
We are aware of this issue and we're currently working on it. There's no way to choose the chain at the given moment. We'll provide a solution eventually.

**Reply by SashaYenchik** [reply]
Have you already provided information on this issue?
